File: ciyuanplus2/app/src/main/java/com/ciyuanplus/mobile/adapter/ExchangeMallAdapter.java
package com.ciyuanplus.mobile.adapter;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ciyuanplus.mobile.R;


public class ExchangeMallAdapter extends RecyclerView.Adapter<ExchangeMallAdapter.ViewHolder> {
    @NonNull
    @Override
    public ExchangeMallAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.exchange_mall_item, parent, false);
        ExchangeMallAdapter.ViewHolder holder = new ExchangeMallAdapter.ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull ExchangeMallAdapter.ViewHolder holder, int position) {
          holder.ll_exchange_mall.setOnClickListener(new View.OnClickListener() {
              @Override
              public void onClick(View v) {
                  if(clickCallBack!=null){
                      clickCallBack.callBack();
                  }
              }
          });
    }

    @Override
    public int getItemCount() {
        return 10;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private final LinearLayout ll_exchange_mall;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ll_exchange_mall = itemView.findViewById(R.id.ll_exchange_mall);
        }
    }
    //定义接口
    private ClickCallBack clickCallBack;

    public void setClickCallBack(ClickCallBack clickCallBack) {
        this.clickCallBack = clickCallBack;
    }

    public interface ClickCallBack {
        void callBack();
    }
}

File: ciyuanplus2/app/src/main/java/com/ciyuanplus/mobile/adapter/ExchangeMallAdapter.kt
package com.ciyuanplus.mobile.adapter

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.graphics.Paint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.ciyuanplus.mobile.R
import com.ciyuanplus.mobile.net.bean.RankedFirstItem
import com.ciyuanplus.mobile.net.response.CommodityListItemRes
import com.ciyuanplus.mobile.net.response.ShopProdListItemRes
import com.ciyuanplus.mobile.utils.Constants
import com.ciyuanplus.mobile.widget.SquareImageView
import com.kris.baselibrary.util.NumberUtil
import com.orhanobut.logger.Logger

import java.util.ArrayList


class ExchangeMallAdapter(var context: Context?, data: List<ShopProdListItemRes.CommodityItem>) :
        BaseQuickAdapter<ShopProdListItemRes.CommodityItem, BaseViewHolder>(R.layout.exchange_mall_item, data) {
    override fun convert(helper: BaseViewHolder?, item: ShopProdListItemRes.CommodityItem?) {

        if (item != null) {


            helper!!.setText(R.id.tv_title, item.name)
                    .setText(R.id.tv_price_now, "￥ ${NumberUtil.getAmountValue(item.price.toDouble() / 100)}")
                    .setText(R.id.tv_price_original, "￥ ${NumberUtil.getAmountValue(item.orgPrice.toDouble() / 100)}")

            helper.getView<TextView>(R.id.tv_price_original).paintFlags = Paint.STRIKE_THRU_TEXT_FLAG
            helper.getView<TextView>(R.id.tv_price_original).paint.isAntiAlias = true

            val list = item.img.split(",")
            Logger.d("图片地址 ${Constants.IMAGE_LOAD_HEADER + list[0]}")

            val options = RequestOptions().placeholder(R.drawable.ic_default_image_007).error(R.mipmap.imgfail).dontAnimate().centerCrop()
            Glide.with(mContext).load(Constants.IMAGE_LOAD_HEADER + list[0]).apply(options).into(helper.getView(R.id.iv_image))
            /*if(item.source==null){
                helper.setVisible(R.id.hot_taobao,false);
                helper.setVisible(R.id.tv_price_original,true)
            } else if(item.source=="taobao"){
                helper.setVisible(R.id.hot_taobao,true);
                helper.setVisible(R.id.tv_price_original,false)
            }*/

        }
    }
}
